include/wlr/types/wlr_region.h

```diff
@@ -0,0 +1,10 @@
+#ifndef _WLR_TYPES_REGION_H
+#define _WLR_TYPES_REGION_H
+
+struct wl_resource;
+
+// Implements the given resource as region.
+// Sets the associated pixman_region32_t as userdata.
+void wlr_region_create(struct wl_resource *res);
```

Comment:
But it cannot know when the region is destroyed. The region is probably only created by the compositor implementation.